core/arch/arm/include/kernel/thread.h

```diff
@@ -285,28 +287,29 @@ int thread_get_id_may_fail(void);
 struct thread_specific_data *thread_get_tsd(void);
 
 /*
- * Sets IRQ status for current thread, must only be called from an
- * active thread context.
+ * Sets foreign interrupts status for current thread, must only be called
+ * from an active thread context.
  *
- * enable == true  -> enable IRQ
- * enable == false -> disable IRQ
+ * enable == true  -> enable foreign interrupts
+ * enable == false -> disable foreign interrupts
  */
-void thread_set_irq(bool enable);
+void thread_set_foreign_intr(bool enable);
 
 /*
- * Restores the IRQ status (in CPSR) for current thread, must only be called
- * from an active thread context.
+ * Restores the foreign interrupts status (in CPSR) for current thread, must
+ * only be called from an active thread context.
  */
-void thread_restore_irq(void);
+void thread_restore_foreign_intr(void);
 
 /*
  * Defines the bits for the exception mask used the the
  * thread_*_exceptions() functions below.
  */
-#define THREAD_EXCP_FIQ	(1 << 0)
-#define THREAD_EXCP_IRQ	(1 << 1)
-#define THREAD_EXCP_ABT	(1 << 2)
-#define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
+#define THREAD_EXCP_FOREIGN_INTR	(ARM32_CPSR_I >> ARM32_CPSR_F_SHIFT)
+#define THREAD_EXCP_NATIVE_INTR		(ARM32_CPSR_F >> ARM32_CPSR_F_SHIFT)
+#define THREAD_EXCP_ALL			(THREAD_EXCP_FOREIGN_INTR	\
+					| THREAD_EXCP_NATIVE_INTR	\
+					| (ARM32_CPSR_A >> ARM32_CPSR_F_SHIFT))
```

Comment:
Hi @zhizhouzh,
We have talked about this. I agree with @jenswi-linaro that the "THREAD_EXCP_FOREIGN_INTR" can replace THREAD_EXCP_IRQ or THREAD_EXCP_FIQ.
Because they are flags for "THREAD", actually, they are not necessarily aligned with the bit definitions with ARM. (I know now they are and optee_os uses them as ARM architecture macros for FIQ/IRQ/ABT in some functions. e.g. `thread_set/get_exceptions`).